rust/feature-flags/src/flag_matching.rs

```diff
@@ -137,7 +224,12 @@ impl FeatureFlagMatcher {
                 return Ok(self.check_rollout(feature_flag, rollout_percentage));
             }
 
-            let target_properties = self.get_target_properties(feature_flag, properties).await?;
+            let target_properties = if feature_flag.get_group_type_index().is_some() {
+                self.get_group_properties(feature_flag).await?
+            } else {
+                self.get_person_properties(feature_flag.team_id, properties)
+                    .await?
+            };
```

Comment:
@neilkakkar I want to make sure my understanding is complete here.

In Django, we have [this bit](https://github.com/PostHog/posthog/blob/master/posthog/models/feature_flag/flag_matching.py#L326-L346), which (if I understand correctly), is saying that any payload properties we want to match the flag conditions against  are either
- group properties (if the `aggregation_group_type_index` on the flag exists) OR
- person property overrides

In other words, we either match group properties, or match person properties, but not both.

Furthermore, I noticed that in Django we explicitly set these values to empty dict if they aren't present on the payload.

My question, then, is:
- is the only reason that create an empty iterable in Django so that we can take advantage of chaining properties using methods like `all`?
- is there any other subtlety that I'm missing here?  

In my `get_group_properties` method (defined [here](https://github.com/PostHog/posthog/pull/24565/files#diff-a0cf62f2ebc7e77042e6ff8843b57769c02013fe43910be22f3f25303bac9720R246-R262), or just keep scrolling), I end up making empty hashmaps so that the types line up, but I'd rather not allocate a map for these properties unless I have to.